xc/common/utils/prjxray_create_place_constraints.py

```diff
@@ -741,6 +753,46 @@ def main():
                 block, vpr_loc, "Constraining clock block {}".format(block)
             )
 
+    # Constrain IDELAYCTRL sites
```

Comment:
I think it is worth documenting what needs have happened prior to this point.  Per https://www.xilinx.com/support/answers/39966.html, IDELAYCTRL replication needs have been completed by the point this script is invoked.  So I completely agree with the assertion at line 766.

The current code handles the unconstrained case (#2 from above) and it handles case #1 from above.  The missing case is #3, where an `IODELAY_GROUP` is defined.

Earlier in the flow (before packing), IDELAYCTRL replication needs to be handled per either the `IODELAY_GROUP` or for the unconstrained case.

Reply:
Added comment explaining the three situations as reported here https://www.xilinx.com/support/answers/39966.html.